src/main/java/com/sngular/kloadgen/extractor/extractors/ProtoBufExtractor.java

```diff
@@ -20,25 +20,50 @@
 import com.squareup.wire.schema.internal.parser.OneOfElement;
 import com.squareup.wire.schema.internal.parser.ProtoFileElement;
 import com.squareup.wire.schema.internal.parser.TypeElement;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.RandomUtils;
 import org.apache.commons.lang3.StringUtils;
 
-public class ProtoBufExtractor {
+public class ProtoBufExtractor implements Extractor {
 
   public static final String ARRAY_POSTFIX = "-array";
 
   public static final String MAP_POSTFIX = "-map";
 
   public static final String UNSUPPORTED_TYPE_OF_VALUE = "Something Odd Just Happened: Unsupported type of value";
 
-  public final List<FieldValueMapping> processSchema(final ProtoFileElement schema) {
+  public final List<FieldValueMapping> processSchema(final Object schema) {
     final List<FieldValueMapping> attributeList = new ArrayList<>();
     final HashMap<String, TypeElement> nestedTypes = new HashMap<>();
-    schema.getTypes().forEach(field -> processField(field, attributeList, schema.getImports(), true, nestedTypes));
+    final ProtoFileElement protoFileElement = ((ProtobufSchema) schema).rawSchema();
+    protoFileElement.getTypes().forEach(field -> processField(field, attributeList, protoFileElement.getImports(), true, nestedTypes));
     return attributeList;
   }
 
+  @Override
+  public final List<FieldValueMapping> processApicurioParsedSchema(final Object schema) {
+    final List<FieldValueMapping> attributeList = new ArrayList<>();
+    final var protoFileElement = ((io.apicurio.registry.utils.protobuf.schema.ProtobufSchema) schema).getProtoFileElement();
+    // TODO: instead of sending null, check if it's needed to create an HashMap with field.getNestedTypes()
```

Comment:
Is there a reason why this change can't be done along with the rest of the PR?

Reply:
It's an old comment and I have no idea of what it speaks about